### DISCUSSION TITLE
API Queries - Liquid User Data

### DISCUSSION THREAD

**Discussion by emrahtokalak** [OrchardCMS/OrchardCore#16521] [Q&A] [answered]
Hello,

I am running a sql query from my application's interface using the api/queries/{name} endpoint. The user is logged into my application. I want to access user information like "{% assign user = User | user_id | users_by_id %}" with liquid in my query. I think the user information in httpcontext is not loaded to liquid running here, what do you suggest I do for this?

**Comment by Piedone** [comment, marked as answer]
Yes, because web APIs need to be authenticated with [OpenID](https://docs.orchardcore.net/en/latest/reference/modules/OpenId/). You can take some inspiration from [our API client project](https://github.com/Lombiq/Orchard-Core-API-Client) too.
